### PR TITLE
Fix all unit tests on macOS

### PR DESCRIPTION
**Pull request by jturcotte**
Also fixes the UNIT_TESTING build on Windows.

### CODE REVIEW

File: csync/src/csync_private.h [hunk 1]
```diff
@@ -207,7 +207,7 @@ __attribute__ ((packed))
 #endif
 ;
 
-void csync_file_stat_free(csync_file_stat_t *st);
+OCSYNC_EXPORT void csync_file_stat_free(csync_file_stat_t *st);
```
Comment (ogoffart) [inline csync/src/csync_private.h]: In Qt there is a  Q_XXX_AUTIOTEST_EXPORT  for this use case.

I'm not saying we should do the same.

**Comment by guruz** [comment]
(Haven't looked at the code actually because I assume @ogoffart did)

File: src/libsync/utility.cpp [hunk 2]
```diff
@@ -277,6 +277,9 @@ bool Utility::fsCasePreserving()
     if( isWindows() || isMac() ) {
         re = true;
     } else {
+#ifndef WITH_TESTING
+        static
```
Comment (jturcotte) [inline src/libsync/utility.cpp]: Fixed the test to run on macOS instead.

File: csync/src/csync_private.h (hunk 1 above)
Comment (jturcotte) [inline csync/src/csync_private.h]: Qt however tries to be tidy with its export, I think we can live without it unless it poses a concrete problem.

**Pull request merged**